ci/integration_tests/src/test_harness/test_runner.rs

```diff
@@ -0,0 +1,79 @@
+extern crate test;
+
+use cosm_orc::{
+    config::cfg::Config, orchestrator::cosm_orc::CosmOrc, profilers::gas_profiler::GasProfiler,
+};
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::panic;
+use std::time::Instant;
+use test::TestDescAndFn;
+
+use super::chain::Chain;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+enum Status {
+    Ok,
+    Fail,
+}
+
+// Runs all integration tests in one thread
+pub fn cosm_orc_test_runner(tests: &[&TestDescAndFn]) {
+    setup();
+
+    println!("running {} integration tests", tests.len());
+
+    let mut passed = 0;
+    let mut failed = 0;
+    let time = Instant::now();
+
+    for &test in tests {
+        if let test::TestFn::StaticTestFn(f) = test.testfn {
+            let res = panic::catch_unwind(f);
+            if res.is_err() {
+                failed += 1;
+            } else {
+                passed += 1;
+            }
+        } else {
+            todo!()
+        }
+
+        // create a clean slate for next test
+        Chain::clear_deploys();
```

Comment:
Cosm-orc will save the contract address when a contract is instantiated (like it does with the code ids when contracts are stored). Each test reuses the same contract IDs but the contract addresses are cleared after each test, to create a clean starting state for each test.